homeassistant/components/reolink/update.py

```diff
@@ -123,15 +132,44 @@ def latest_version(self) -> str | None:
 
         return new_firmware.version_string
 
+    @property
+    def in_progress(self) -> bool:
+        """Update installation progress."""
+        return self._host.api.sw_upload_progress(self._channel) < 100
+
+    @property
+    def update_percentage(self) -> int:
+        """Update installation progress."""
+        return self._host.api.sw_upload_progress(self._channel)
+
     @property
     def supported_features(self) -> UpdateEntityFeature:
         """Flag supported features."""
         supported_features = UpdateEntityFeature.INSTALL
         new_firmware = self._host.api.firmware_update_available(self._channel)
         if isinstance(new_firmware, NewSoftwareVersion):
             supported_features |= UpdateEntityFeature.RELEASE_NOTES
+            supported_features |= UpdateEntityFeature.PROGRESS
         return supported_features
 
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        if self._installing or self._cancel_update is not None:
+            return True
+        return super().available
+
+    def version_is_newer(self, latest_version: str, installed_version: str) -> bool:
+        """Return True if latest_version is newer than installed_version."""
+        try:
+            installed = SoftwareVersion(installed_version)
+            latest = SoftwareVersion(latest_version)
+        except ReolinkError:
+            # when the online update API returns a unexpected string
+            return True
```

Comment:
If the installed version is an unexpected string, then we will always show that an update is available, even if that could be wrong. Can we make it more stable?

Reply:
It can't be wrong:
Either the update comes from the reolink.com server, in which case the version string comes from a NewSoftwareVersion class which inherits from SoftwareVersion class, so you can never get this error.

Or the update comes from the API of the camera, in which case a weird version string could trigger this exceptions. But in that case the API of the camera is directly telling you there is an update. So we can trust the internal camera logic that their actually is a new update. It will then also be able to install using the camera API (online install).

Reply:
In almost all cases updates will come from the reolink.com server.